docs/source/cli-reference/basic-options.rst

```diff
@@ -9,6 +9,61 @@
 
 ----
 
+``-dir, --additional_directories`` Options
+------------------------------------------
+
+    .. admonition:: Dependency
+
+        The option ``-dir, --additional_directories`` is a sub-option of and requires the option
+        ``--license``.
+
+    The ``-dir, --additional_directories`` option allows the user to include additional directories
+    of licenses to use in license detection.
+
+    This command only needs to be run once for each set of additional directories; in all subsequent
+    runs of Scancode with the same directories all the licenses in the directories will be cached.
+
+    The directory structure should look something like this::
+
+        licenses/
+        ├── privateLicense1/
+        │   ├── license/
+        │   │   ├── privateLicense1.LICENSE
+        │   │   └── privateLicense1.yml
+        │   └── rule/
+        │       ├── privateLicense1.RULE
+        │       └── privateLicense1.yml
+        └── privateLicense2/
+            ├── license/
+            │   ├── privateLicense2.LICENSE
+            │   └── privateLicense2.yml
+            └── rule/
+                ├── privateLicense2.RULE
+                └── privateLicense2.yml
+
+    A scan example using the ``-dir, --additional_directories PATH`` option with a single directory::
+
+        scancode -clpieu --json-pp output.json samples -dir /home/user/external_licenses/license1
+
+    You can also include multiple directories like so::
+
+        scancode -clpieu --json-pp output.json samples -dir /home/user/external_licenses/external1 -dir /home/user/external_licenses/external2
+
+    If you want to continue running scans with ``/home/user/external_licenses/external1`` and ``/home/user/external_licenses/external2``,
+    you can omit the ``-dir`` option in subsequent scans and they will still be included. ::
+
+        scancode -clpieu --json-pp output.json samples
+
+    However, if you wanted to run a scan with a new set of directories, such as ``home/user/external_licenses/external1``
```

Comment:
In this case, we'd have to run 
`
scancode -clpieu --json-pp output.json samples -dir /home/user/external_licenses/external1 -dir /home/user/external_licenses/external2 -dir /home/user/external_licenses/external3`

Right now the code checks if the directories passed in from the command line are a subset of the directories included in the cache, and if they aren't, it recreates the index with the input directories. So in this case, initially, `external1` and `external2` would be in the index. If we ran 
`scancode -clpieu --json-pp output.json samples -dir /home/user/external_licenses/external1 -dir /home/user/external_licenses/external2 -dir /home/user/external_licenses/external3
`, then `external1`, `external2`, and `external3` would be in the index.

If we ran `scancode -clpieu --json-pp output.json samples -dir /home/user/external_licenses/external3`, only `external3` would be in the index.

I implemented it this way since there isn't a way to "remove" licenses from the index, so I thought that if we always kept on adding licenses to the index each time someone used `-dir`, the index could get really large. Do you think I should implement it such that they always stay in the index?

Reply:
I think what you did makes sense. Thanks for clarifying!